SU2_CFD/src/numerics_direct_turbulent.cpp

```diff
@@ -1222,42 +1222,320 @@ void CSourcePieceWise_TurbSST::ComputeResidual(su2double *val_residual, su2doubl
   beta_blended = F1_i*beta_1 + (1.0 - F1_i)*beta_2;
   
   if (dist_i > 1e-10) {
-    
-    /*--- Production ---*/
-    
-    diverg = 0.0;
-    for (iDim = 0; iDim < nDim; iDim++)
-      diverg += PrimVar_Grad_i[iDim+1][iDim];
-    
-    pk = Eddy_Viscosity_i*StrainMag_i*StrainMag_i - 2.0/3.0*Density_i*TurbVar_i[0]*diverg;
-    pk = min(pk,20.0*beta_star*Density_i*TurbVar_i[1]*TurbVar_i[0]);
-    pk = max(pk,0.0);
-    
-    zeta = max(TurbVar_i[1], StrainMag_i*F2_i/a1);
-    pw = StrainMag_i*StrainMag_i - 2.0/3.0*zeta*diverg;
-    pw = max(pw,0.0);
-    
-    val_residual[0] += pk*Volume;
-    val_residual[1] += alfa_blended*Density_i*pw*Volume;
-    
-    /*--- Dissipation ---*/
-    
-    val_residual[0] -= beta_star*Density_i*TurbVar_i[1]*TurbVar_i[0]*Volume;
-    val_residual[1] -= beta_blended*Density_i*TurbVar_i[1]*TurbVar_i[1]*Volume;
-    
-    /*--- Cross diffusion ---*/
-    
-    val_residual[1] += (1.0 - F1_i)*CDkw_i*Volume;
-    
-    /*--- Implicit part ---*/
-    
-    val_Jacobian_i[0][0] = -beta_star*TurbVar_i[1]*Volume;
-    val_Jacobian_i[0][1] = -beta_star*TurbVar_i[0]*Volume;
-    val_Jacobian_i[1][0] = 0.0;
-    val_Jacobian_i[1][1] = -2.0*beta_blended*TurbVar_i[1]*Volume;
+
+   /*--- Production ---*/
+
+   diverg = 0.0;
+   for (iDim = 0; iDim < nDim; iDim++)
+     diverg += PrimVar_Grad_i[iDim+1][iDim];
+   
+   /* if using UQ methodolgy, calculate production using perturbed Reynolds stress matrix */
+
+   if (config->GetUsing_UQ()){
+     SetReynoldsStressMatrix(TurbVar_i[0]);
+     SetPerturbedRSM(TurbVar_i[0], config);
+     SetPerturbedStrainMag(TurbVar_i[0]);
+     pk = Eddy_Viscosity_i*PerturbedStrainMag*PerturbedStrainMag
+     - 2.0/3.0*Density_i*TurbVar_i[0]*diverg;
+   }
+   else {
+     pk = Eddy_Viscosity_i*StrainMag_i*StrainMag_i - 2.0/3.0*Density_i*TurbVar_i[0]*diverg;
+   }
+
+
+   pk = min(pk,20.0*beta_star*Density_i*TurbVar_i[1]*TurbVar_i[0]);
```

Comment:
One thing you and I discussed some time ago was exposing the production limiter as a config parameter. Some later variants of SST call for small changes to that value. Are you interested in adding this here?

Reply:
I think I might do a different pull request for that, just so it is treated as a separate issue